ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiMutationConverter.kt

```diff
@@ -40,14 +41,17 @@ internal object LoiMutationConverter {
 
     map.put(LoiConverter.JOB_ID, mutation.jobId)
 
-    mutation.location
-      .map { toGeoPoint(it) }
-      .ifPresent { point: GeoPoint -> map.put(LoiConverter.LOCATION, point) }
-
-    val geometry: MutableMap<String, Any> = HashMap()
-    geometry[LoiConverter.GEOMETRY_COORDINATES] = toGeoPointList(mutation.polygonVertices)
-    geometry[LoiConverter.GEOMETRY_TYPE] = LoiConverter.POLYGON_TYPE
-    map.put(LoiConverter.GEOMETRY, geometry)
+    when (mutation.geometry) {
+      is Point -> map.put(LoiConverter.LOCATION, toGeoPoint(mutation.geometry))
+      is Polygon -> {
+        val geometryMap: MutableMap<String, Any> = HashMap()
+        geometryMap[LoiConverter.GEOMETRY_COORDINATES] =
+          toGeoPointList(mutation.geometry.shell.vertices)
+        geometryMap[LoiConverter.GEOMETRY_TYPE] = LoiConverter.POLYGON_TYPE
+        map.put(LoiConverter.GEOMETRY, geometryMap)
+      }
+      else -> {}
```

Comment:
Can we add a TODO here?

Reply:
For the `else` branch? In this case it's working as intended since this `when` only performs side-effects--the default side-effect is to do nothing.